packages/core/src/enqueue_links/enqueue_links.ts

```diff
@@ -148,15 +185,38 @@ export async function enqueueLinks(options: EnqueueLinksOptions): Promise<QueueO
         ]),
     }));
 
-    const extraPseudoUrls = [];
+    const {
+        requestQueue,
+        limit,
+        urls,
+        pseudoUrls,
+        globs,
+        regexps,
+        transformRequestFunction,
+    } = options;
+
+    let urlPatternObjects: UrlPatternObject[] = [];
+
+    if (pseudoUrls?.length) {
+        log.deprecated('`pseudoUrls` option is deprecated, use `globs` or `regexps` instead');
+        urlPatternObjects = urlPatternObjects.concat(constructRegExpObjectsFromPseudoUrls(pseudoUrls));
+    }
+
+    if (globs?.length) {
+        urlPatternObjects = urlPatternObjects.concat(constructGlobObjectsFromGlobs(globs));
+    }
+
+    if (regexps?.length) {
+        urlPatternObjects = urlPatternObjects.concat(constructRegExpObjectsFromRegExps(regexps));
+    }
 
     if (options.baseUrl) {
         switch (options.strategy ?? EnqueueStrategy.SameSubdomain) {
             case EnqueueStrategy.SameSubdomain:
                 // We need to get the origin of the passed in domain in the event someone sets baseUrl
                 // to a url like https://example.com/deep/default/path and one of the found urls is an
                 // absolute relative path (/path/to/page)
-                extraPseudoUrls.push(`${new URL(options.baseUrl).origin}/[.*]`);
+                urlPatternObjects.push({ regexp: purlToRegExp(`${new URL(options.baseUrl).origin}/[.*]`) });
```

Comment:
Can't we convert this to native regexes / globs now, instead of using purl? (or should we handle that in a future PR)

Reply:
Updated :+1:

Reply:
This is not correct, as you are not escaping special characters in the origin, and it will definitely contain things like dots, question marks or plus signs.

Reply:
Or do you mean `purlToRegExp` is not correct? But I haven't changed the logic - i.e. the hole `purlToRegExp` is the same as it was before...

Reply:
Went for `${new URL(options.baseUrl).origin}/**` glob and not for `${new URL(options.baseUrl).origin}/**/*` because first option would allow to not have anything after `[domain]/` while second with `**/*` would expect something here -> not the same behaviour as before.

Also for `EnqueueStrategy.SameHostname` `${url.origin.replace(baseUrlHostname, *.${baseUrlHostname})}/**` with `**/*` in the end would not allow urls with different prefix, but without deeper levels. E.g. for `www.apify.com` it would not allow something like `sdk.apify.com` or `help.apify.com` and would expect something after `.com`.